site/content/en/docs/build-clusters.md

```diff
@@ -0,0 +1,181 @@
+---
+title: "GKE Build Clusters"
+description: >
+  Sometimes you might want to run all of your jobs in a GKE cluster that is separate from Prow (perhaps running in your own GCP project). What you need is a dedicated "build cluster".
+weight: 3
+---
+
+*Note: This page discusses build clusters that use GKE. Technically speaking, a
+build cluster could be any Kubernetes cluster (not just GKE) because the only
+thing Prow needs is the ability to authenticate as a Kubernetes Service Account
+with cluster-admin role permissions to the build cluster.*
+
+## Overview
+
+By default Prow will schedule jobs in the same Kubernetes cluster as itself, but
+under the `test-pods` namespace. Setting up a separate build cluster will allow
```

Comment:
This is not necessarily true. For example "default" cluster in k8s prow is the build cluster located in `k8s-prow-builds` project instead of `k8s-prow` project. Consider something like:

> By default Prow will schedule jobs in the cluster that maps to kubeconfig alias "default" (Imagine running `kubectl config set-context "default" --cluster=<CLUSTER_CONTEXT>`, the <CLUSTER_CONTEXT> could be any cluster)

Reply:
Speaking of this, I think this document should also include infos related to context aliases. For example a GKE cluster named `cluster-foo` in project `project-bar` under `us-central1` could be named something totally unrelated, such as `the-most-awesome-build-cluster`. And whenever a user wants to configure something for this cluster they should use `the-most-awesome-build-cluster` instead of `cluster-foo`

Reply:
I don't want to make this doc any longer. Let me try rephrasing and/or deleting the sentence you mentioned.

Reply:
if you meant to not extending this PR I would agree. Otherwise imo the `alias` vs real `cluster_name` is where our users got tripped most, and it's worth documented

Reply:
> could be named something totally unrelated

Where is this definition created? I want to add a link to an example.

Reply:
> > could be named something totally unrelated
> 
> Where is this definition created? I want to add a link to an example.

See https://github.com/kubernetes/test-infra/blob/408f66a2a9199fbcaafc55aa35f72a49a06da26c/gencred/cmd/gencred/main.go#L117, this is the `alias`

Reply:
Oh interesting. I was expecting a declarative YAML config somewhere, but it is defined in gencred first. So I: guess it's accurate to say that context aliases start out from gencred (makes sense to me, because kubeconfig seems to be the source of truth WRT how prow is able to talk to other build clusters, and as I now know, the token(s) in the kubeconfig need to be refreshed by gencred).